src/main/java/seedu/address/model/Model.java

```diff
@@ -254,4 +256,21 @@
      */
     void sortFriendList(Comparator<Friend> comparator);
 
+    /**
+     * Checks if accommodation object date is within the model's travel plan start and end date.
+     * @return false if accommodation start date and end date is within travel plan start date and end date.
+     */
+    boolean isValidAccommodationDate(WanderlustDate startDate, WanderlustDate endDate);
+
+    /**
+     * Checks if activity object date time is within the model's travel plan start and end date.
+     * @return true if activity date is within travel plan start date and end date range.
+     */
+    boolean isValidActivityDate(WanderlustDateTime activityDateTime);
+
+    /**
+     * Checks if activity object date time is within the model's travel plan start and end date.
+     * @return true if activity date is within travel plan start date and end date range.
+     */
+    boolean isValidActivityDate(WanderlustDateTime activityDateTime, WanderlustDate startDate, WanderlustDate endDate);
```

Comment:
I think the JavaDoc for this can put like, see isValidActivityDate() something like that? I not sure tho.

Reply:
Think this will be considered as duplicate?

Reply:
What you mean?

Reply:
I'm just concern about if this will be considered duplicates code. I think it's OK ba.